modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/quantity_selector/QuantitySelector.js

```diff
@@ -12,246 +12,122 @@
  * details.
  */
 
+import {ClaySelect} from '@clayui/form';
 import ClayIcon, {ClayIconSpriteContext} from '@clayui/icon';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
-import React, {createRef, useEffect, useState} from 'react';
+import React, {useEffect, useState} from 'react';
 
-import throttle from '../../utilities/throttle';
-
-const THROTTLE_TIMEOUT = 1000;
+import Datalist from '../datalist/Datalist';
+import {generateOptions, getInputStyle} from './utils/index';
 
 function QuantitySelector(props) {
-	const [currentQuantity, setCurrentQuantity] = useState(
-		!!props.minQuantity && props.minQuantity > props.quantity
-			? props.minQuantity
-			: props.quantity
-	);
-	const [nextAvailable, setNextAvailable] = useState(
-		currentQuantity + props.multipleQuantity <= props.maxQuantity
-	);
-	const [prevAvailable, setPrevAvailable] = useState(
-		currentQuantity - props.multipleQuantity >= props.minQuantity
-	);
-	const [isThrottling, setIsThrottling] = useState(false);
+	const [selectedQuantity, setSelectedQuantity] = useState(props.quantity);
+
+	const quantitiesList = generateOptions(props.settings);
+
+	const updateQuantity = (q) => {
+		if (q !== null) {
+			setSelectedQuantity(q);
+			try {
+				props.updateQuantity(q);
+			}
+			catch (e) {
+				// eslint-disable-next-line no-console
+				console.log('stand alone component');
```

Comment:
The fact that you have to suppress the lint here is a sign that something is wrong. We should be logging only in very rare circumstances [as described here](https://github.com/liferay/liferay-frontend-guidelines/blob/master/dxp/debugging.md#logging).